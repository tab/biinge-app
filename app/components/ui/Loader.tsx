import React from "react"
import { View, ActivityIndicator } from "react-native"
import { BlurView } from "@react-native-community/blur"

import { APP_APPEARANCE_DARK, APP_APPEARANCE_LIGHT } from "config"
import { loadingStyles } from "styles"
import colors from "styles/colors"

type Props = {
  dark: boolean
}

const LoaderComponent = ({ dark }: Props) => {
  return (
    <View style={loadingStyles.root}>
      <BlurView
        style={loadingStyles.blur}
        blurType={dark ? APP_APPEARANCE_DARK : APP_APPEARANCE_LIGHT}
        blurAmount={15}
        reducedTransparencyFallbackColor={colors.darkBlack}
      />
      <ActivityIndicator
        animating={true}
        size="small"
        color={colors.white}
      />
    </View>
  )
}

LoaderComponent.defaultProps = {
  dark: true,
}

export default LoaderComponent
