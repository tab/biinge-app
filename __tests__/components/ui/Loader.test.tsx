import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Loader from "components/ui/Loader"

describe("Loader", () => {
  it("renders correctly", () => {
    const tree = render(<Loader />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with dark", () => {
    const tree = render(<Loader dark />)
    expect(tree).toMatchSnapshot()
  })
})
