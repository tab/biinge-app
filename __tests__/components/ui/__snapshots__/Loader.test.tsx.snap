// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
    }
  }
>
  <BlurView
    blurAmount={15}
    blurType="light"
    reducedTransparencyFallbackColor="#000"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    size="small"
  />
</View>
`;

exports[`Loader renders correctly with dark 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
    }
  }
>
  <BlurView
    blurAmount={15}
    blurType="dark"
    reducedTransparencyFallbackColor="#000"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    size="small"
  />
</View>
`;
